### PR TITLE
Add Switch Case flow-control block with per-case execution branches

### PR DESCRIPTION
**Pull request by patricknihranz**
## Description

Adds `roboflow_core/switch_case@v1`, a flow-control block that routes execution to one of several branches by matching an input value against case keys — a multi-way alternative to chaining Continue If blocks.

**Block semantics**
- `value` (selector or literal) is string-coerced and matched against the keys of `cases: Dict[str, StepSelector]` (e.g. `{"red": "$steps.on_red"}`); optional `case_insensitive` toggle.
- On match, execution continues to that case's step only; otherwise `default_next_steps` runs; if that is empty the branch terminates.
- A manifest validator rejects duplicate targets across `cases` + `default_next_steps` (each target step may be referenced at most once per flow-control step — engine constraint) and case-key collisions under case-insensitive matching, with readable errors instead of the engine's compile-time one.
- `cases` is a flat dict because selector discovery only finds step selectors one level deep (`schema_parser.py` ignores nested references).

**Execution engine changes (required for per-case branching)**
1. `graph_constructor.py`: selectors held in **dict** properties now get a unique execution branch per key (`Branch[$steps.switch -> cases[red]]`). Previously all selectors in one property shared a single branch, so selecting one case would have activated every case target. **List** properties (`next_steps`) keep the shared-branch behavior — ContinueIf semantics are unchanged.
2. `execution_data_manager/manager.py`: deduplicate branch names when registering non-SIMD flow-control masks. This fixes a **pre-existing bug**: any non-SIMD flow-control step with two or more `next_steps` crashed with "Attempted to re-register maks for execution branch" (reproducible on main with plain ContinueIf; regression test added).

**UI metadata**: `cases` is marked `always_visible` so the builder renders it as a primary field. Companion frontend PR in roboflow/roboflow adds per-case connection handles in the workflow builder.

## Testing

- 22 unit tests (`tests/workflows/unit_tests/core_steps/control_flow/test_switch_case.py`): manifest validation + run() behavior incl. coercion edge cases (`True`→"True", `None`→"None", `1.0`→"1.0").
- 5 integration tests (`test_workflow_with_switch_case.py`): case match / default / terminate routing through `ExecutionEngine`, plus compile-time duplicate-target failure.
- 2 regression tests for the multi-`next_steps` ContinueIf fix (`test_workflow_with_flow_control.py`).
- Full `tests/workflows/unit_tests/execution_engine/` suite passes (542 total in the affected areas).
- Manually verified end-to-end through a local HTTP server (`/workflows/run`) and from the workflow builder UI.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by patricknihranz** [comment]
Companion frontend PR: roboflow/roboflow#12428

File: inference/core/workflows/core_steps/flow_control/switch_case/v1.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+                )
+        return self
+
+    @classmethod
+    def describe_outputs(cls) -> List[OutputDefinition]:
+        return []
+
+    @classmethod
+    def get_execution_engine_compatibility(cls) -> Optional[str]:
+        return ">=1.3.0,<2.0.0"
```
Comment (dkosowski87) [inline inference/core/workflows/core_steps/flow_control/switch_case/v1.py]: Two things here:
1. We need to document the changes in `graph_constructor` and `execution_data_manager`. This should be done by bumping the execution engine version in `inference/core/workflows/execution_engine/v1/core.py` and providing a proper entry in the [CHANGELOG](docs/workflows/execution_engine_changelog.md). There are some version assertions also in test, you can find them by searching by the current EE version, we need to update them.
2. The `">=1.3.0,<2.0.0"` will be incorrect as the block depends on the changes in the execution engine, the min has to be also bumped to the new version.

Comment (patricknihranz) [reply inference/core/workflows/core_steps/flow_control/switch_case/v1.py]: Addressed in 47b48ce:

- Bumped `EXECUTION_ENGINE_V1_VERSION` 1.10.1 → **1.11.0** (per-case dict execution branches + non-SIMD mask dedup fix).
- Added a changelog entry under `v1.11.0` describing both engine changes.
- Raised the block's `get_execution_engine_compatibility` floor to `>=1.11.0,<2.0.0`.
- Updated the EE version assertions in `test_workflow_endpoints.py` and `hosted_platform_tests/test_workflows.py` (the assertions in `test_core.py` / `test_compilation_of_incompatible_plugin.py` reference the constant symbolically, so they track automatically).

**Review by dkosowski87 — APPROVED** [review]
🥇
